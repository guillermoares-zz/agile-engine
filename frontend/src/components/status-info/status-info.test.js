import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer'
import StatusInfo, {STATUS_INFO_KIND} from './status-info';
import Alert from 'react-bootstrap/Alert';


describe('StatusInfo', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<StatusInfo/>)

    expect(component.toJSON()).toMatchSnapshot()
  })

  describe('Rendering', () => {
    it('should show info messages', () => {
      const message = "some message"
      const kind = STATUS_INFO_KIND.INFO
      
      let wrapper = shallow(<StatusInfo message={message} kind={kind}/>);

      expect(wrapper.find(Alert).props().children).toEqual(message)
      expect(wrapper.find(Alert).props().variant).toEqual('info')
    });

    it('should show error messages', () => {
      const message = "some error message"
      const kind = STATUS_INFO_KIND.ERROR

      let wrapper = shallow(<StatusInfo message={message} kind={kind}/>);

      expect(wrapper.find(Alert).props().children).toEqual(message)
      expect(wrapper.find(Alert).props().variant).toEqual('danger')
    });

    it('should show success messages', () => {
      const message = "some success message"
      const kind = STATUS_INFO_KIND.SUCCESS

      let wrapper = shallow(<StatusInfo message={message} kind={kind}/>);

      expect(wrapper.find(Alert).props().children).toEqual(message)
      expect(wrapper.find(Alert).props().variant).toEqual('success')
    });
    
  })

})