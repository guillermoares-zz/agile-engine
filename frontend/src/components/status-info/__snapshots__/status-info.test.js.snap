// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusInfo snapshot renders 1`] = `
<div
  className="fade alert show"
  role="alert"
/>
`;
